backend/fms_core/models/step_history.py

```diff
@@ -19,7 +20,8 @@
 class StepHistory(TrackedModel):
     study = models.ForeignKey(Study, on_delete=models.PROTECT, related_name="StepHistory", help_text="Study associated to the process measurement.")
     step_order = models.ForeignKey(StepOrder, on_delete=models.PROTECT, related_name="StepHistory", help_text="Step order in the study that is associated to the process measurement.")
-    process_measurement = models.ForeignKey(ProcessMeasurement, on_delete=models.PROTECT, related_name="StepHistory", help_text="Process measurement associated to the study step.")
+    process_measurement = models.ForeignKey(ProcessMeasurement, null=True, blank=True, on_delete=models.PROTECT, related_name="StepHistory", help_text="Process measurement associated to the study step.")
+    sample = models.ForeignKey(Sample, on_delete=models.PROTECT, related_name="StepHistory", help_text="Source sample that completed the step.")
```

Comment:
what is the purpose of this sample field?

Reply:
When there is no process_measurement the association between a StepHistory and a specific sample cannot be made otherwise... It replaces process_measurement.source_sample to provide the sample mapping...